docs/SocialLogin.md

```diff
@@ -0,0 +1,368 @@
+    provider would redirect the browser after the user authorized our app to access his data
+
+* Implicit Flow
+
+    * https://datatracker.ietf.org/doc/html/rfc6749
+    * https://oauth.net/2/grant-types/implicit/
+    * This flow was supposed to be for browser-based single-page apps, but is now legacy for security reasons
+    * This could theoretically be implemented without needing a `client_secret` browser-side
+
+* PKCE Flow
+
+    - https://datatracker.ietf.org/doc/html/rfc7636
+    - https://oauth.net/2/pkce/
+    - This could theoretically be implemented mostly in the browser without needing a `client_secret`
+    accessible there
+    - This flow is not supported by all social login providers we take into consideration - it's supported
+    on GitLab, but not on GitHub
+
+Given the limittations of the flows above, we are limitted with regards to options if we want all 3 to work.
+Login with GitHub requires an implementation of the Authorization Code Flow, which implies more zot server side code. 
+
+## Data-Path login
+
+The above is mostly applicable for zot UI side.
+For Data-Path, we can use personal token from GitHub. We could investigate further to see how we could obtain
+GitLab or Google tokens.
+Given the GitHub token is already generated, we can jump straight to validating it on zot side.
+
+## Observations on validating an existing token and user permissions
+
+* Currently we don't have a way to identify what social login provider was used when we receive a token by just inspecting it's string.
+The GitHub token is prefixed with `gh` (`gho_*` for oauth2 token and `ghp_*` for personal tokens),
+but there is no similar convention for other providers.
+* Also we don't have a way to decode the tokens, in order to verify it's validity/permissions
+* If the user has a choice of multiple providers, we would need to check in the backend by validating it against
+all configured providers.
+
+Someone suggested custom headers on requests from frontend to specify token type (github/google/gitlab), but
+this would only work for zot UI, and not for the data path.
+
+In order to validate a token there is no straight forward way to make a standard call, it depends on the provider.
+For example GitHub provides a non-rate-limitted API for validating oauth2 `gho_*` tokens,
+but there is no such API for personal `ghp_*` tokens.
+
+This means we have to call another rate-limitted API to check if login with said token works on GitHun, or
+the social login provider in general.
+We need to find a way around this.
+
+One option is to keep the tokens in memory or in a database (just the valid one, or also the invalid ones?).
+This way we won't hit the rate limitting, but we also do not know if the token has expired in the meantime.
+A possible solution would be to validate the tokens once, and then use a different token which we emit ourselves or cookie - at
```

Comment:
There won't be a need to validate against a Social Login provider endpoint if the token is generated on Zot server side.
As explained in the above comment,  period of validity is available in the in-memory map / DB so no extra query for that. We'll just use the refresh_token to generate access_token, if that's the case.

Reply:
Not all tokens are generated on Zot server side. There's also personal access tokens.